scalafix-core/shared/src/main/scala/scalafix/internal/util/TypeSyntax.scala

```diff
@@ -1,138 +1,451 @@
 package scalafix.internal.util
 
+import java.io.PrintWriter
+import java.io.StringWriter
+import org.langmeta.internal.semanticdb._
 import scala.meta._
+import scala.meta.internal.{semanticdb3 => s}
 import scala.reflect.ClassTag
+import scala.util.control.NonFatal
 import scalafix._
 import scalafix.internal.util.SymbolOps.SymbolType
-import scalafix.util.SymbolMatcher
-import scalafix.util.TreeExtractors._
 import org.langmeta.semanticdb.Symbol
+import scala.meta.internal.semanticdb3.SingletonType.{Tag => x}
+import scala.meta.internal.semanticdb3.SymbolInformation.{Kind => k}
+import scala.meta.internal.semanticdb3.SymbolInformation.{Property => p}
+import scala.meta.internal.semanticdb3.Type.{Tag => t}
+import TypeSyntax._
+import scalapb.GeneratedMessage
 
 object TypeSyntax {
-  def prettify(tpe: Type, ctx: RuleCtx, shortenNames: Boolean)(
-      implicit index: SemanticdbIndex): (Type, Patch) = {
+  def prettify(
+      stpe: s.Type,
+      ctx: RuleCtx,
+      shortenNames: Boolean,
+      pos: Position)(implicit index: SemanticdbIndex): Option[(Type, Patch)] = {
+    try {
+      val table = index.asInstanceOf[EagerInMemorySemanticdbIndex].table
```

Comment:
Woah, this cast looks scary.

Reply:
Seconded, why do we know more than the compiler here?

Reply:
The `SemanticdbIndex` trait should only have a single implementation, I've kept the abstract trait to separate the public API from the implementation. I admit this is not an ideal situation but the alternative is to expose implementation details in the public API. We have the same situation in other places, for example for linter suppression. `private[scalafix]` does not bite enough IMO compared to forcing a cast.